include/mbedtls/mbedtls_config.h

```diff
@@ -2065,6 +2065,28 @@
  */
 #define MBEDTLS_AESNI_C
 
+/**
+ * \def MBEDTLS_AESCE_C
+ *
+ * Enable AES crypto extension support on Arm64.
+ *
+ * Module:  library/aesce.c
+ * Caller:  library/aes.c
+ *
+ * Requires: MBEDTLS_HAVE_ASM, MBEDTLS_AES_C
+ *
+ * \note The code uses Neon intrinsics, so \c CFLAGS must be set to a minimum
+ * of \c -march=armv8-a+crypto .
+ *
+ * \warning `MBEDTLS_SHA512_USE_A64_CRYPTO_*` should be disabled when enabled
```

Comment:
They might or might not work depending on the compiler version and command line switches. Since some users might be happy with those options and able to get away with having them both. I  think this could justify not enforcing it.

Reply:
Then shouldn't this text be more explanatory?

But actually, isn't this warning wrong? The issue is with setting compiler flags needed for `MBEDTLS_SHA512_USE_A64_CRYPTO_IF_PRESENT` and running the resulting code on a system without the SHA3 crypto extension (because compiler optimisation uses the SHA3 feature set elsewhere) - or am I misinterpreting this warning (which would also suggest that it needs updating!)

If the library is built with `MBEDTLS_SHA512_USE_A64_CRYPTO_ONLY`, and the compiler flags set accordingly, that's not a problem, because "only" means "this will definitely be run on a system with the SHA3 crypto extension"

Reply:
The problem is that if both are present then with -O3 a SHA3 instruction will be compiled into the AES code. If this runs on a platform with crypto extension but no SHA3 extension, the AES detection will think that we have support and then the code chokes on the SHA3 instruction.

Reply:
I think we're saying the same thing - probably writing our comments at the same time!

I don't think this warning is the best way to express the potential issue to a user of the library

Reply:
> If the library is built with MBEDTLS_SHA512_USE_A64_CRYPTO_ONLY, and the compiler flags set accordingly, that's not a problem, because "only" means "this will definitely be run on a system with the SHA3 crypto extension"

Agree if `runtime detection` can be disabled at project level. 

For time being, we do not have an option to disable runtime detection of sha512 and AESCE together. In theory, `MBEDTLS_SHA512_USE_A64_CRYPTO_ONLY` and `MBEDTLS_AES_USE_A64_CRYPTO_IF_PRESENT` can be enabled together.

Reply:
> `MBEDTLS_SHA512_USE_A64_CRYPTO_ONLY` and `MBEDTLS_AES_USE_A64_CRYPTO_IF_PRESENT` can be enabled together

That's not what the warning currently says

Reply:
Unfortunately I can't find a better way of writing this warning than

```
 * \warning If the target architecture is set to something that includes the
            SHA3 feature (e.g. `-march=armv8.2-a+sha3`), for example because
            `MBEDTLS_SHA512_USE_A64_CRYPTO_IF_PRESENT` is desired, some
            compilers generate code for `MBEDTLS_AESCE_C` that includes
            instructions only present with the (optional) SHA3 feature. This
            will lead to an undefined instruction exception if the code is run
            on a CPU without that feature.
```

Reply:
I change `some compilers` to `compilers`. The optimization is reasonable if we tell compiler `sha3` is supported.